.github/actions/chart-test-recipes/action.yml

```diff
@@ -0,0 +1,179 @@
+        TEST_INGRESS_HOST: ${{ steps.vars.outputs.ingress-host }}
+      run: |
+        timeout 5m task -d $TEST_SCENARIOS_DIR/chart-full-setup setup.pre
+
+        echo "Extra values from workflow:"
+        echo "${{ inputs.extra-values }}" > /tmp/extra-values-file.yaml
+        cat /tmp/extra-values-file.yaml
+
+    - name: 🌟 Setup Camunda chart 🌟
+      shell: bash
+      env:
+        TEST_CHART_FLOW: ${{ inputs.flow }}
+        TEST_HELM_EXTRA_ARGS: >-
+          --set global.ingress.host=${{ steps.vars.outputs.ingress-host }} --values /tmp/extra-values-file.yaml
+        TEST_PROMETHEUS_HELM_EXTRA_ARGS: "--set host=${{ steps.vars.outputs.ingress-host }}" # this variable was previously set on ${{ steps.ingress.outputs.host }} but no such steps exists, check if it sill works
+      run: |
+        task -d $TEST_SCENARIOS_DIR/chart-full-setup setup.exec
+
+    - name: Post setup
+      shell: bash
+      run: timeout 5m task -d $TEST_SCENARIOS_DIR/chart-full-setup setup.post
+
+    - name: Pre Upgrade
+      shell: bash
+      if: inputs.flow == 'upgrade'
+      run: task -d $TEST_SCENARIOS_DIR/chart-full-setup upgrade.pre
+
+    - name: 🌟 Upgrade Camunda chart 🌟
+      shell: bash
+      if: inputs.flow == 'upgrade'
+      env:
+        TEST_HELM_EXTRA_ARGS: >-
+          --set global.ingress.host=${{ steps.vars.outputs.ingress-host }} --values /tmp/extra-values-file.yaml
+      run: task -d $TEST_SCENARIOS_DIR/chart-full-setup upgrade.exec
+
+    - name: Update GitHub deployment status
+      uses: bobheadxi/deployments@648679e8e4915b27893bd7dbc35cb504dc915bc8 # v1
+      with:
+        step: finish
+        token: ${{ steps.generate-github-token.outputs.token }}
+        status: ${{ job.status }}
+        deployment_id: ${{ steps.deployment.outputs.deployment_id }}
+        env_url: https://${{ steps.vars.outputs.ingress-host }}
+        env: ${{ steps.vars.outputs.identifier }}
+        ref: ${{ inputs.caller-git-ref }}
+
+    - name: ⭐️ Run Preflight TestSuite ⭐️
+      shell: bash
+      if: inputs.test-enabled == 'true'
+      run: timeout 10m task -d $TEST_SCENARIOS_DIR/chart-full-setup test.preflight
+
+    - name: ⭐️ Run Core TestSuite ⭐️
+      shell: bash
+      if: inputs.test-enabled == 'true'
+      run: timeout 20m task -d $TEST_SCENARIOS_DIR/chart-full-setup test.core
+
+    - name: 🚨 Get failed Pods info 🚨
+      if: failure()
+      uses: ./.github/actions/failed-pods-info
+
+    - name: Cleanup GitHub deployment
+      if: always() && (inputs.persistent == 'false' || inputs.type != 'kubernetes')
+      uses: bobheadxi/deployments@648679e8e4915b27893bd7dbc35cb504dc915bc8 # v1
+      with:
+        step: delete-env
+        token: ${{ steps.generate-github-token.outputs.token }}
+        env: ${{ steps.vars.outputs.identifier }}
+        ref: ${{ inputs.caller-git-ref }}
+
+    - name: Cleanup test namespace
+      shell: bash
+      if: always() && (inputs.persistent == 'false' || inputs.type != 'kubernetes')
```

Comment:
I get it that you didn't write it but I'm wondering about why `inputs.type != 'kubernetes'` is done?
We previously create both ns and deployment nevertheless the type but not we have an exception in place for the cleanup?

Reply:
@leiicamundi This is to make sure that when the persistent flag is set to true, only the gke environment is persisted and not the openshift environment.